blocks/video/video.js

```diff
@@ -0,0 +1,374 @@
+
+  setupPlayer(videoContainer, {
+    url: config.videoUrl,
+    autoplay: config.isAutoPlay,
+    hasCustomPlayButton: true,
+  });
+
+  container.append(article);
+}
+
+function decorateHeroBlock(block, config) {
+  const container = document.createElement('div');
+  container.classList.add('video-hero');
+
+  const content = document.createElement('div');
+  content.classList.add('video-hero-content');
+
+  if (config.title) {
+    const title = document.createElement('h1');
+    title.classList.add('video-hero-title');
+    title.textContent = config.title;
+    content.append(title);
+  }
+
+  if (config.description) {
+    const description = document.createElement('p');
+    description.classList.add('video-hero-description');
+    description.textContent = config.description;
+    content.append(description);
+  }
+
+  if (config.button) {
+    const button = document.createElement('button');
+    button.classList.add('video-hero-button');
+    button.textContent = config.button;
+    content.append(button);
+  }
+
+  container.append(content);
+
+  block.innerHTML = '';
+  block.append(container);
+
+  setupPlayer(container, {
+    url: config.videoUrl,
+    autoplay: config.isAutoPlay,
+    hasCustomPlayButton: true,
+  });
+}
+
+function decorateVideoCards(block, config) {
+  const gridContainer = document.createElement('ul');
+  gridContainer.classList.add('video-card-grid');
+
+  block.innerHTML = '';
+  block.append(gridContainer);
+
+  config.cards.forEach((videoConfig) => {
+    const gridItem = document.createElement('li');
+    gridItem.classList.add('video-card-grid-item');
+    gridContainer.append(gridItem);
+
+    decorateVideoCard(gridItem, videoConfig);
+  });
+}
+
+function closeModal() {
+  const modal = document.querySelector('.video-modal');
+  modal.querySelector('.video-container').innerHTML = '';
+
+  // eslint-disable-next-line no-use-before-define
+  window.removeEventListener('click', handleOutsideClick);
+  // eslint-disable-next-line no-use-before-define
+  window.removeEventListener('keydown', handleEscapeKey);
+
+  modal.style.display = 'none';
+  document.body.style.overflow = '';
+}
+
+function handleOutsideClick(event) {
+  const modal = document.querySelector('.video-modal');
+  if (event.target === modal) {
+    closeModal();
+  }
+}
+
+function handleEscapeKey(event) {
+  if (event.key === 'Escape') {
+    closeModal();
+  }
+}
+
+async function openModal(config) {
+  await loadVideoJs();
+
+  const modal = document.querySelector('.video-modal');
+  const container = modal.querySelector('.video-container');
+  setupPlayer(container, {
+    url: config.videoUrl,
+    autoplay: false,
+    controls: true,
+    bigPlayButton: true,
+  });
+
+  window.addEventListener('click', handleOutsideClick);
+  window.addEventListener('keydown', handleEscapeKey);
+
+  modal.style.display = '';
+  document.body.style.overflow = 'hidden';
+}
+
+function createModal() {
+  const modal = document.createElement('div');
```

Comment:
please use https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog instead of div